### PR TITLE
GH Action tests for backend

### PR DESCRIPTION
**Pull request by mikeshultz**
Adds GH action to run backend tests and fixes some things along the way.

### CODE REVIEW

File: .github/workflows/backend.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
+
+      - name: Install dependencies
+        run: yarn
+
+      - name: Build frontend distribution for backend
+        run: yarn run build:dist
```
Comment (franckc) [inline .github/workflows/backend.yml:45]: This is a slow step. Is that necessary?
It takes a while to build (several minutes on my host).

**Review by franckc — APPROVED** [review]
Great! Thanks for adding this.

One question inline otherwise LGTM

File: .github/workflows/backend.yml (hunk 1 above)
Comment (mikeshultz) [reply .github/workflows/backend.yml:45]: Yeah... Some of the tests reliy on having the shop build available.  We might be able to optimize later by adding this as artifacts.  I don't think the tests really care that the build is up to date.

**Comment by franckc** [comment]
Maybe for the tests we only build 1 theme by passing it as an argument to the script?
https://github.com/OriginProtocol/dshop/blob/master/shop/buildThemes.sh#L31

**Pull request merged**